fondant/dataset.py

```diff
@@ -31,56 +31,75 @@ def __init__(self, manifest: Manifest):
         }
 
     def _load_subset(
-        self, name: str, fields: t.List[str], index: t.Optional[Index] = None
+        self, subset_name: str, fields: t.List[str], index: t.Optional[Index] = None
     ) -> dd.DataFrame:
-        # get subset from the manifest
-        subset = self.manifest.subsets[name]
-        # get remote path
+        """
+        Function that loads the subset
+        Args:
+            subset_name: the name of the subset to load
+            fields: the fields to load from the subset
+            index: the manifest index to filter the subset on.
+        Returns:
+            The subset as a dask dataframe
+        """
+
+        subset = self.manifest.subsets[subset_name]
         remote_path = subset.location
-
-        # add index fields
         index_fields = list(self.manifest.index.fields.keys())
         fields = index_fields + fields
 
-        logger.info(f"Loading subset {name} with fields {fields}...")
+        logger.info(f"Loading subset {subset_name} with fields {fields}...")
 
-        df = dd.read_parquet(
-            remote_path,
-            columns=fields,
-        )
+        subset_df = dd.read_parquet(remote_path, columns=fields)
```

Comment:
Is this also a Dask task?

Reply:
It returns a dask dataframe object (lazily loaded). The `to_parquet` method defines a task to write to a parquet file so both are separate.